components/core/src/clp_s/search/Output.hpp

```diff
@@ -18,372 +18,42 @@
 #include "ast/StringLiteral.hpp"
 #include "clp_search/Query.hpp"
 #include "OutputHandler.hpp"
+#include "QueryRunner.hpp"
 #include "SchemaMatch.hpp"
 
 using namespace simdjson;
 using namespace clp_s::search::clp_search;
 
 namespace clp_s::search {
-class Output : public FilterClass {
+class Output {
 public:
-    Output(SchemaMatch& match,
-           std::shared_ptr<ast::Expression> expr,
-           std::shared_ptr<ArchiveReader> archive_reader,
-           std::shared_ptr<TimestampDictionaryReader> timestamp_dict,
+    Output(std::shared_ptr<SchemaMatch> const& match,
+           std::shared_ptr<ast::Expression> const& expr,
+           std::shared_ptr<ArchiveReader> const& archive_reader,
            std::unique_ptr<OutputHandler> output_handler,
            bool ignore_case)
-            : m_archive_reader(std::move(archive_reader)),
-              m_schema_tree(m_archive_reader->get_schema_tree()),
-              m_schemas(m_archive_reader->get_schema_map()),
-              m_match(match),
+            : m_query_runner(match, expr, archive_reader, ignore_case),
+              m_archive_reader(std::move(archive_reader)),
```

Comment:
```suggestion
              m_archive_reader(archive_reader),
```
I think it might be better to not `std::move` `archive_reader` since we also pass it to `m_query_runner`, so reordering could lead to a null pointer dereference.